src/compiler/utilities.ts

```diff
@@ -1667,6 +1667,27 @@ namespace ts {
         return SyntaxKind.FirstKeyword <= token && token <= SyntaxKind.LastKeyword;
     }
 
+    // Some keywords are TypeScript only, so they should not be parsed as considered as
+    // keywords in JavaScript
+    export function isJSKeyword(token: SyntaxKind): boolean {

```

Comment:
If you keep it around, use `isSolelyJavaScriptKeyword`.

Additionally, you might want to note that these keywords are language version dependent.